src/nplinker/pairedomics/downloader.py

```diff
@@ -720,11 +702,37 @@ def _download_metabolomics_zipfile(self, gnps_task_id):
                               path=os.path.join(self.project_file_cache,
                                                 'spectra'))
 
+    def _log_gnps_format(self):
         if self._is_new_gnps_format(self.project_file_cache):
             logger.info('Found NEW GNPS structure')
         else:
             logger.info('Found OLD GNPS structure')
 
+    def _load_gnps_data(self, gnps_task_id) -> zipfile.ZipFile:
+
+        self.metabolomics_zip = os.path.join(self.project_download_cache,
+                                             'metabolomics_data.zip')
+
+        # Try read from cache
+        if os.path.exists(self.metabolomics_zip):
+            logger.info('Found existing metabolomics_zip at {}'.format(
+                self.metabolomics_zip))
+            try:
+                mbzip = zipfile.ZipFile(self.metabolomics_zip)
+                return mbzip
+            except zipfile.BadZipFile as bzf:
+                logger.info(
+                    'Invalid metabolomics zipfile found, will download again!')
+                os.unlink(self.metabolomics_zip)
+        else:
+            url = _generate_gnps_download_url(gnps_task_id)
```

Comment:
why not using `GNPSDownloader`?

Reply:
I haven't yet started replacing the actual NPLinker code with the refactored part - will add that to TODOs